README.md

```diff
@@ -270,35 +270,128 @@ For iOS, the repeating notification should land soon. It has already been merged
 
 ```
-import PushNotificationAndroid from 'react-native-push-notification'
-
-(function() {
-  // Register all the valid actions for notifications here and add the action handler for each action
-  PushNotificationAndroid.registerNotificationActions(['firstAction','secondAction']); // here provide action id’s from previous step
-  DeviceEventEmitter.addListener('notificationActionReceived', function(action){
-    console.log ('Notification action received: ' + action);
-    const info = JSON.parse(action.dataJSON);
-    if (info.action == 'firstAction') {
-      // Do work pertaining to Accept action here
-    } else if (info.action == 'secondAction') {
-      // Do work pertaining to Reject action here
+package your.app.package.name;
+
+import android.app.ActivityManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+import com.facebook.react.HeadlessJsTaskService;
+
+import java.util.List;
+
+public class NotificationAcitonHandlerReceiver extends BroadcastReceiver {
+
+  @Override
+  public void onReceive(Context context, Intent intent) {
+    Intent serviceIntent = new Intent(context, NotificationActionHandlerService.class);
+    serviceIntent.putExtras(intent.getExtras());
+    context.startService(serviceIntent);
+
+    // Dismiss the notification popup.
+    Bundle bundle = intent.getBundleExtra("notification");
+    NotificationManager manager = (NotificationManager) context.getSystemService(context.NOTIFICATION_SERVICE);
+    int notificationID = Integer.parseInt(bundle.getString("id"));
+    manager.cancel(notificationID);
+
+    if (!isAppOnForeground((context))) {
+      HeadlessJsTaskService.acquireWakeLockNow(context);
+    }
+  }
+
+  private boolean isAppOnForeground(Context context) {
+    /**
+     We need to check if app is in foreground otherwise the app will crash.
+     http://stackoverflow.com/questions/8489993/check-android-application-is-in-foreground-or-not
+     **/
+    ActivityManager activityManager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+    List<ActivityManager.RunningAppProcessInfo> appProcesses =
+        activityManager.getRunningAppProcesses();
+    if (appProcesses == null) {
+      return false;
+    }
+    final String packageName = context.getPackageName();
+    for (ActivityManager.RunningAppProcessInfo appProcess : appProcesses) {
+      if (appProcess.importance ==
+          ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND &&
+          appProcess.processName.equals(packageName)) {
+        return true;
+      }
     }
-    // Add all the required actions handlers
-  });
-})();
+    return false;
+  }
+}
+```
+
+### 5) Modify AndroidManifest.xml
+Add `service` and `receiver` with `intent-filter` to your `AndroidMainfest.xml` file.
+
+```
+<service android:name=".NotificationActionHandlerService" />
+<receiver android:name=".NotificatitonAcitonHandlerReceiver"
```

Comment:
Typo: NotificatitonAcitonHandlerReceiver => NotificatitonActionHandlerReceiver

Reply:
This is fixed in followup commit